mobx/decorators.js

```diff
@@ -39,10 +40,43 @@ export function settable(target, property, descriptor) {
  * reference. This will use the similarly named `@observable.ref` decorator in the core MobX API.
  */
 export function bindable(target, property, descriptor) {
-    return settable(target, property, observable(target, property, descriptor));
+    return createBindable(target, property, descriptor, false);
+
 }
+
 bindable.ref = function(target, property, descriptor) {
-    return settable(target, property, observable.ref(target, property));
+    return createBindable(target, property, descriptor, true);
 };
 
+//-----------------
+// Implementation
+//-----------------
+function createBindable(target, name, descriptor, isRef) {
+
+    // 1) set up a set function, if one does not exist.  This was the original side-effect of
+    // bindable and still used for backward compatibility by HoistBase.setBindable.
+    const setterName = 'set' + upperFirst(name);
+    if (!target.hasOwnProperty(setterName)) {
+        const value = function(v) {
+            this[name] = v;
+        };
+        Object.defineProperty(target, setterName, {value});
```

Comment:
How does this ensure the `setXXX` function is an @action and thus works with mobx strict mode?

Reply:
the set function it delegates to on line 61 is actually an action.

Reply:
Ah yes, of course - this is using the setter defined below 👍